relay-server/src/actors/project_cache.rs

```diff
@@ -446,6 +447,34 @@ impl Services {
     }
 }
 
+/// Tracks how many projects are accessed every second.
+#[derive(Debug)]
+struct AccessTracker {
+    keys: HashSet<ProjectKey>,
+    last_reset: Instant,
+}
+
+impl AccessTracker {
+    fn new() -> Self {
+        Self {
+            keys: Default::default(),
+            last_reset: Instant::now(),
+        }
+    }
+
+    fn track(&mut self, project_key: ProjectKey) {
+        if self.last_reset.elapsed() > Duration::from_secs(1) {
+            let count = self.keys.len();
+            // When we get < 1 access per second, this will log results from the past.
```

Comment:
Assuming <1 access per second happens when `keys` is empty, why does this log results from the past?